src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/magical/EnchantmentRune.java

```diff
@@ -0,0 +1,141 @@
+import org.bukkit.entity.Player;
+import org.bukkit.event.player.PlayerDropItemEvent;
+import org.bukkit.inventory.ItemStack;
+
+import me.mrCookieSlime.Slimefun.Lists.RecipeType;
+import me.mrCookieSlime.Slimefun.Objects.Category;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.SimpleSlimefunItem;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.SlimefunItem;
+import me.mrCookieSlime.Slimefun.Objects.handlers.ItemDropHandler;
+import me.mrCookieSlime.Slimefun.SlimefunPlugin;
+import me.mrCookieSlime.Slimefun.api.Slimefun;
+import me.mrCookieSlime.Slimefun.api.SlimefunItemStack;
+
+/**
+ * This {@link SlimefunItem} allows you to enchant any enchantable {@link ItemStack} with a random
+ * {@link Enchantment}. It is also one of the very few utilisations of {@link ItemDropHandler}.
+ *
+ * @author Linox
+ *
+ * @see ItemDropHandler
+ * @see Enchantment
+ *
+ */
+public class EnchantmentRune extends SimpleSlimefunItem<ItemDropHandler> {
+
+    private static final double RANGE = 1.5;
+    private final Map<Material, List<Enchantment>> applicableEnchantments = new EnumMap<>(Material.class);
+
+    public EnchantmentRune(Category category, SlimefunItemStack item, RecipeType recipeType, ItemStack[] recipe) {
+        super(category, item, recipeType, recipe);
+
+        for (Material mat : Material.values()) {
+            List<Enchantment> enchantments = new ArrayList<>();
+            for (Enchantment enchantment : Enchantment.values()) {
+                if (enchantment == Enchantment.BINDING_CURSE || enchantment == Enchantment.VANISHING_CURSE) continue;
+                if (enchantment.canEnchantItem(new ItemStack(mat))) enchantments.add(enchantment);
+            }
+            applicableEnchantments.put(mat, enchantments);
+        }
+    }
+
+    @Override
+    public ItemDropHandler getItemHandler() {
+        return (e, p, item) -> {
+            if (isItem(item.getItemStack())) {
+
+                if (!Slimefun.hasUnlocked(p, this, true)) {
+                    return true;
+                }
+
+                Slimefun.runSync(() -> activate(p, e, item), 20L);
+
+                return true;
+            }
+            return false;
+        };
+    }
+
+    private void activate(Player p, PlayerDropItemEvent e, Item item) {
+        // Being sure the entity is still valid and not picked up or whatsoever.
+        if (!item.isValid()) {
+            return;
+        }
+
+        Location l = item.getLocation();
+        Collection<Entity> entites = l.getWorld().getNearbyEntities(l, RANGE, RANGE, RANGE, this::findCompatibleItem);
+        Optional<Entity> optional = entites.stream().findFirst();
+
+        if (optional.isPresent()) {
+            Item entity = (Item) optional.get();
+            ItemStack target = entity.getItemStack();
+
+            List<Enchantment> enchantmentList = applicableEnchantments.getOrDefault(target.getType(), new ArrayList<>());
+            if (enchantmentList.isEmpty()) return;
```

Comment:
Hmmm, well you are returning anyway when the list is empty...
So you could just do a simple get and do a null check instead of creating a new ArrayList and then immediately discarding it.